src/XProfile/VirtoCommerce.XProfile/Authorization/CanEditOrganizationAuthorizationHandler.cs

```diff
@@ -0,0 +1,164 @@
+                foreach (var userName in deleteUserCommand.UserNames)
+                {
+                    if (allowDelete)
+                    {
+                        var user = await _userManager.FindByNameAsync(userName);
+                        allowDelete = await HasSameOrganizationAsync(currentContact, user.MemberId);
+                    }
+                }
+
+                result = allowDelete;
+            }
+            else if (context.Resource is UpdateContactAddressesCommand updateContactAddressesCommand)
+            {
+                result = updateContactAddressesCommand.ContactId == GetUserId(context);
+                if (!result && currentContact!= null)
+                {
+                    result = await HasSameOrganizationAsync(currentContact, updateContactAddressesCommand.ContactId);
+                }
+            }
+            else if (context.Resource is UpdateContactCommand updateContactCommand && currentContact != null)
+            {
+                result = updateContactCommand.Id == currentContact.Id;
+                if (!result)
+                {
+                    result = await HasSameOrganizationAsync(currentContact, updateContactCommand.Id);
+                }
+            }
+            else if (context.Resource is UpdateOrganizationCommand updateOrganizationCommand && currentContact != null)
+            {
+                result = currentContact.Organizations.Contains(updateOrganizationCommand.Name);
+            }
+            else if (context.Resource is UpdateRoleCommand updateRoleCommand)
+            {
+                //Can be checked only with platform permission
+                result = true;
+            }
+            else if (context.Resource is UpdateUserCommand updateUserCommand && currentContact != null)
+            {
+                result = updateUserCommand.MemberId == currentContact.Id;
+                if (!result)
+                {
+                    result = await HasSameOrganizationAsync(currentContact, updateUserCommand.Id);
+                }
+            }
+
+            if (result)
+            {
+                context.Succeed(requirement);
+            }
+            else
+            {
+                context.Fail();
+            }
+        }
+
+        private static string GetUserId(AuthorizationHandlerContext context)
+        {
+            //TODO use ClaimTypes instead of "name"
+            return context.User.FindFirstValue("name");
+        }
+        private async Task<bool> HasSameOrganizationAsync(Contact currentContact, string contactId)
+        {
+            var contact = await GetCustomerAsync(contactId) as Contact;
+            return currentContact.Organizations.Intersect(contact?.Organizations ?? Array.Empty<string>()).Any();
+        }
+
+        //TODO: DRY violation in many places in this solution. Move to abstraction to from multiple boundaries
+        protected virtual async Task<Member> GetCustomerAsync(string customerId)
```

Comment:
Need to replace to `IMemberResolver` that was added in this commit https://github.com/VirtoCommerce/vc-module-experience-api/commit/6be8eee65c2b30a92c55828a48c5c40f89ad2008

Reply:
Will change this when merging PR with this commit again after the release.